### PR TITLE
Adds self-diagnosis messages for various borg interactions

### LINKED ISSUE #13989: Borgs get no info prompts for various borg-related things
#### Description of issue

There's no prompt for having your cover locked, opened, for cell changes or when having upgrades applied.
Also no signal for being blown apart from suddenly being gibbed

### PR DESCRIPTION
**Pull request by SonixApache**
Adds self-diagnosis messages for various borg related things, like
- interface locking and unlocking
- cover opening and closing
- component additions and removals
- radio modifications
- cell changes
- upgrade installations

all of these only when the self-diagnosis component is functional 

Revision 1:
- Added a helper proc to slightly reduce resource usage
- added messages for when components are destroyed *except for the self-diagnosis unit*
- fixed up the message saying which broken component is removed
- added failure / success message to upgrade installations

image
image

resolves #13989
tested
🆑 
 - rscadd: Borgs now get some messages regarding what's happening to their body and components, as long as their self-diagnosis unit is working.

### CODE REVIEW

**Comment by ghost** [comment]
Very nice, quality of life!

**Comment by SkowronX** [comment]
Coders should focus more on things like this

**Comment by SonixApache** [comment]
Should be good to go.

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -818,6 +821,8 @@
 				W.forceMove(null)
 
 				to_chat(usr, "<span class='notice'>You install the [W.name].</span>")
+				if(src.can_diagnose())
```
Comment (ghost) [inline code/modules/mob/living/silicon/robot/robot.dm]: Sweet jesus why are you using src.?

**Review by ghost — CHANGES_REQUESTED** [review]
remove src.

File: code/modules/mob/living/silicon/robot/robot.dm (hunk 1 above)
Comment (SonixApache) [reply code/modules/mob/living/silicon/robot/robot.dm]: i have no idea

**Review by ghost — APPROVED** [review]

File: code/modules/mob/living/silicon/robot/robot.dm (hunk 1 above)
Comment (Exxion) [reply code/modules/mob/living/silicon/robot/robot.dm]: It completely doesn't matter at all. I still don't know why people got so upset about it for a few days.

Comment (JustSumBody) [reply code/modules/mob/living/silicon/robot/robot.dm]: NONSTANDARD FORMATTING CHANGE AAAAAAAAAAAAAAAA

**Comment by SonixApache** [comment]
neat

**Pull request merged**